src/lucky_cli/task.cr

```diff
@@ -60,6 +68,131 @@ abstract class LuckyCli::Task
     end
   end
 
+  macro positional_arg(arg_name, description, required = nil, format = nil, to = nil)
+    {% PARSER_OPTS << arg_name %}
+    @{{ arg_name.id }} : {% if to %}Array(String){% else %}String{% end %} | Nil
+
+    def set_opt_for_{{ arg_name.id }}(args : Array(String))
+      {% if to %}
+        value = args[@positional_arg_count..-1]
+      {% else %}
+        value = args[@positional_arg_count]?
+      {% end %}
+      {% if format %}
+      matches = value.is_a?(Array) ? value.all?(&.=~({{ format }})) : value =~ {{ format }}
+      if matches
+        @{{ arg_name.id }} = value
+      else
+        raise "Invalid format for {{ arg_name.id }}. It should match {{ format }}"
+      end
+      {% else %}
+        @{{ arg_name.id }} = value
+      {% end %}
+      @positional_arg_count += 1
+    end
+
+    def {{ arg_name.id }}
+      {% if required %}
+        if @{{ arg_name.id }}.nil?
+          raise <<-ERROR
+          {{ arg_name.id }} is required, but no value was passed.
+          ERROR
+        end
+        @{{ arg_name.id }}.not_nil!
+      {% else %}
+        @{{ arg_name.id }}
+      {% end %}
+    end
+  end
+
+  macro arg(arg_name, description, shortcut = nil, required = nil, format = nil)
+    {% PARSER_OPTS << arg_name %}
+    @{{ arg_name.id }} : String?
+
+    def set_opt_for_{{ arg_name.id }}(unused_args : Array(String))
+      {% if shortcut %}
+      option_parser.on(
+        "{{ shortcut.id }} {{ arg_name.stringify.upcase.id }}",
+        "--{{ arg_name.id.stringify.underscore.gsub(/_/, "-").id }}={{ arg_name.id.stringify.upcase.id }}",
+        {{ description }}
+      ) do |value|
+        value = value.strip
+        {% if format %}
+        if value =~ {{ format }}
+          @{{ arg_name.id }} = value
+        else
+          raise "Invalid format for {{ arg_name.id }}. It should match {{ format }}"
+        end
+        {% else %}
+          @{{ arg_name.id }} = value
+        {% end %}
+      end
+      {% else %}
+      option_parser.on(
+        "--{{ arg_name.id.stringify.underscore.gsub(/_/, "-").id }}={{ arg_name.id.stringify.upcase.id }}",
+        {{ description }}
+      ) do |value|
+        value = value.strip
+        {% if format %}
+        if value =~ {{ format }}
+          @{{ arg_name.id }} = value
+        else
+          raise "Invalid format for {{ arg_name.id }}. It should match {{ format }}"
+        end
+        {% else %}
+          @{{ arg_name.id }} = value
+        {% end %}
+      end
+      {% end %}
```

Comment:
I noticed both option parsers had the same body, so maybe they could be dried up a little.
```suggestion
      option_parser.on(
        {% if shortcut %}"{{ shortcut.id }} {{ arg_name.stringify.upcase.id }}",{% end %}
        "--{{ arg_name.id.stringify.underscore.gsub(/_/, "-").id }}={{ arg_name.id.stringify.upcase.id }}",
        {{ description }}
      ) do |value|
        value = value.strip
        {% if format %}
        if value =~ {{ format }}
          @{{ arg_name.id }} = value
        else
          raise "Invalid format for {{ arg_name.id }}. It should match {{ format }}"
        end
        {% else %}
          @{{ arg_name.id }} = value
        {% end %}
      end
```

Reply:
Good call 👍